phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
Don't like it too much, but if you want, we might:

1. Deprecate is_enableable
2. Make not_enableable_reason return either string/array like now, or boolean; any "false" return (false, '', array()) would mean the extension is enableable; any "true" value would mean not enableable
3. Have a default implementation of not_enableable_reason as !is_enableable
4. Change the core to use the new function with reverse logic.